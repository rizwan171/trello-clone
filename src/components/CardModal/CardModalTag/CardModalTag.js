import React from "react";
import { FiX } from "react-icons/fi";

const CardModalTag = ({ name }) => {
  return (
    <div className="flex items-center gap-1 py-1 px-2 rounded-sm bg-blue-600 font-semibold text-white">
      <p>{name}</p>
      {/* <FiX className="cursor-pointer" /> */}
    </div>
  );
};

export default CardModalTag;
