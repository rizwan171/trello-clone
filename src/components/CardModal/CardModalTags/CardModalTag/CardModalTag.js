import React, { useEffect, useRef } from "react";

const CardModalTag = ({ tag }) => {
  const tagRef = useRef();

  useEffect(() => {
    tagRef.current.style.background = tag.colour;
  }, []);

  return (
    <div ref={tagRef} className="flex items-center gap-1 py-1 px-2 rounded-sm font-semibold text-white">
      <p>{tag.name}</p>
    </div>
  );
};

export default CardModalTag;