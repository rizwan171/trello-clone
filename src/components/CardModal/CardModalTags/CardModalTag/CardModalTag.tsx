import { useEffect, useRef } from "react";
import { CardModalTagProps } from "../../../../types/components/CardModalTagProps";

const CardModalTag = ({ tag }: CardModalTagProps) => {
  const tagRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (tagRef.current) {
      tagRef.current.style.background = tag.colour;
    }
  }, []);

  return (
    <div ref={tagRef} className="flex items-center gap-1 py-1 px-2 rounded-sm font-semibold text-white">
      <p>{tag.name}</p>
    </div>
  );
};

export default CardModalTag;
