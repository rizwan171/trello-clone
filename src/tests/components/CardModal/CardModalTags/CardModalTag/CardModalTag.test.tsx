import CardModalTag from "../../../../../components/CardModal/CardModalTags/CardModalTag/CardModalTag";
import { renderWithProviders } from "../../../../utils/renderUtils";

describe("CardModalTag", () => {
  it("should render successfully", () => {
    const tag = {
      id: "1",
      name: "Tag 1",
      colour: "#FF3244",
    };

    const view = renderWithProviders(<CardModalTag tag={tag} />);
    expect(view.asFragment()).toMatchSnapshot();
  });
});
