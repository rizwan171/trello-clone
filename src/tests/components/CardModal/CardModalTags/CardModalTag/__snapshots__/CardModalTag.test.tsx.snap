// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardModalTag should render successfully 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-1 py-1 px-2 rounded-sm font-semibold text-white"
    style="background: rgb(255, 50, 68);"
  >
    <p>
      Tag 1
    </p>
  </div>
</DocumentFragment>
`;
